sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -559,6 +559,25 @@ case class BinaryPredicate(override val prettyName: String, left: Expression, ri
   }
 }
 
+trait CollationCheck {
```

Comment:
Rename to `CollationCompatibilityCheck` maybe to make it more clear what it does?